internal/applier/bundledeployment_applier.go

```diff
@@ -1,35 +1,50 @@
 package applier
 
 import (
+	"context"
+
 	rukpakv1alpha1 "github.com/operator-framework/rukpak/api/v1alpha1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 
 	platformtypes "github.com/timflannagan/platform-operators/api/v1alpha1"
+	"github.com/timflannagan/platform-operators/internal/sourcer"
 )
 
 const (
 	plainProvisionerID    = "core-rukpak-io-plain"
 	registryProvisionerID = "core-rukpak-io-registry"
 )
 
-func NewBundleDeployment(o *platformtypes.Operator, image string) *rukpakv1alpha1.BundleDeployment {
+func Apply(ctx context.Context, o *platformtypes.Operator, c client.Client, source sourcer.Bundle) (*rukpakv1alpha1.BundleDeployment, error) {
 	bd := &rukpakv1alpha1.BundleDeployment{}
 	bd.SetName(o.GetName())
-
 	controllerRef := metav1.NewControllerRef(o, o.GroupVersionKind())
-	bd.SetOwnerReferences([]metav1.OwnerReference{*controllerRef})
 
-	bd.Spec = buildBundleDeployment(image)
-	return bd
+	labels := map[string]string{
+		"core.olm.io/package":          o.Spec.Package.Name,
+		"core.olm.io/version":          source.Version,
+		"core.olm.io/source-name":      source.SourceInfo.Name,
+		"core.olm.io/source-namespace": source.SourceInfo.Namespace,
+	}
```

Comment:
This propagates this relevant metadata returned by the sourcing layer, and while it's nice to have this auxiliary data available for visibility and to fuel list queries, we really should be bubbling this up to the operator's status.